services/keyword_service/keyword_test.go

```diff
@@ -69,27 +88,41 @@ func (s *KeywordServiceDbTestSuite) TestGetAllWithInvalidUser() {
 
 func (s *KeywordServiceDbTestSuite) TestSaveWithValidParams() {
 	keywordList := []string{"Hazard", "Ronaldo", "Neymar", "Messi", "Mbappe"}
-	result, err := s.keywordService.Save(keywordList)
+	err := s.keywordService.Save(keywordList)
 
-	assert.Equal(s.T(), 5, len(result))
 	assert.Equal(s.T(), nil, err)
 }
 
-func (s *KeywordServiceDbTestSuite) TestSaveWithValidInvalidUser() {
+func (s *KeywordServiceDbTestSuite) TestSaveWithInvalidUser() {
 	keywordList := []string{"Hazard", "Ronaldo", "Neymar", "Messi", "Mbappe"}
 	keywordService := KeywordService{}
-	result, err := keywordService.Save(keywordList)
+	err := keywordService.Save(keywordList)
 
 	assert.Equal(s.T(), "something went wrong, please try again", err.Error())
-	assert.Equal(s.T(), nil, result)
 }
 
 func (s *KeywordServiceDbTestSuite) TestSaveWithEmptyKeywordList() {
 	keywordList := []string{}
-	result, err := s.keywordService.Save(keywordList)
+	err := s.keywordService.Save(keywordList)
 
 	assert.Equal(s.T(), "invalid data", err.Error())
-	assert.Equal(s.T(), nil, result)
+}
+
+func (s *KeywordServiceDbTestSuite) TestSaveWithEnqueueJobError() {
+	enqueueScrapingJobFunc := google_scraping_service.EnqueueScrapingJob
+	google_scraping_service.EnqueueScrapingJob = func(savedKeyword models.Keyword) error {
+		return errors.New("mock enqueue scraping job error")
+	}
+	defer func() { google_scraping_service.EnqueueScrapingJob = enqueueScrapingJobFunc }()
```

Comment:
SO that's the whole mocking setup, right?

Reply:
Yes it is. The idea is override the function for testing purpose.